File: app/src/main/java/forpdateam/ru/forpda/ui/fragments/profile/adapters/WarningsAdapter.java
package forpdateam.ru.forpda.ui.fragments.profile.adapters;

import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.core.content.ContextCompat;

import forpdateam.ru.forpda.R;
import forpdateam.ru.forpda.entity.remote.profile.ProfileModel;
import forpdateam.ru.forpda.ui.views.adapters.BaseAdapter;
import forpdateam.ru.forpda.ui.views.adapters.BaseViewHolder;

/**
 * Created by radiationx on 15.09.17.
 */

class WarningsAdapter extends BaseAdapter<ProfileModel.Warning, WarningsAdapter.WarningHolder> {
    @Override
    public WarningHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        return new WarningHolder(inflateLayout(parent, R.layout.profile_sub_item_warning));
    }

    @Override
    public void onBindViewHolder(WarningHolder holder, int position) {
        holder.bind(getItem(position));
    }

    class WarningHolder extends BaseViewHolder<ProfileModel.Warning> {
        private final TextView title;
        private final TextView date;
        private final TextView content;

        WarningHolder(View itemView) {
            super(itemView);
            title = itemView.findViewById(R.id.item_title);
            date = itemView.findViewById(R.id.item_date);
            content = itemView.findViewById(R.id.item_content);
        }

        @Override
        public void bind(ProfileModel.Warning item) {
            title.setText(item.getTitle());
            date.setText(item.getDate());
            content.setText(item.getContent());
            switch (item.getType()) {
                case POSITIVE:
                    title.setTextColor(ContextCompat.getColor(title.getContext(), R.color.md_green_400));
                    break;
                case NEGATIVE:
                    title.setTextColor(ContextCompat.getColor(title.getContext(), R.color.md_red_400));
                    break;
            }
        }
    }
}

File: app/src/main/java/forpdateam/ru/forpda/ui/fragments/profile/adapters/WarningsAdapter.kt
package forpdateam.ru.forpda.ui.fragments.profile.adapters

import android.view.View
import android.view.ViewGroup
import by.kirich1409.viewbindingdelegate.viewBinding
import forpdateam.ru.forpda.R
import forpdateam.ru.forpda.databinding.ProfileSubItemWarningBinding
import forpdateam.ru.forpda.entity.remote.profile.ProfileModel
import forpdateam.ru.forpda.entity.remote.profile.ProfileModel.WarningType
import forpdateam.ru.forpda.extensions.getColorFromAttr
import forpdateam.ru.forpda.ui.fragments.profile.adapters.WarningsAdapter.WarningHolder
import forpdateam.ru.forpda.ui.views.adapters.BaseAdapter
import forpdateam.ru.forpda.ui.views.adapters.BaseViewHolder

/**
 * Created by radiationx on 15.09.17.
 */
internal class WarningsAdapter : BaseAdapter<ProfileModel.Warning, WarningHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): WarningHolder {
        return WarningHolder(inflateLayout(parent, R.layout.profile_sub_item_warning))
    }

    override fun onBindViewHolder(holder: WarningHolder, position: Int) {
        holder.bind(getItem(position))
    }

    inner class WarningHolder(itemView: View) : BaseViewHolder<ProfileModel.Warning>(itemView) {
        private val binding by viewBinding<ProfileSubItemWarningBinding>()

        override fun bind(item: ProfileModel.Warning) {
            binding.itemTitle.text = item.title
            binding.itemDate.text = item.date
            binding.itemContent.text = item.content
            val color = when (item.type) {
                WarningType.Positive -> {
                    binding.itemTitle.context.getColor(R.color.md_green_400)
                }

                WarningType.Negative -> {
                    binding.itemTitle.context.getColor(R.color.md_red_400)
                }

                WarningType.Unknown -> {
                    binding.itemTitle.context.getColorFromAttr(R.attr.default_text_color)
                }
            }
            binding.itemTitle.setTextColor(color)
        }
    }
}